clients/main.go

```diff
@@ -0,0 +1,223 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/go-openapi/loads/fmts"
+	"github.com/go-openapi/spec"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	swaggerURL = "https://raw.githubusercontent.com/iron-io/functions/master/docs/swagger.yml"
+	rootTmpDir = "tmp"
+)
+
+func main() {
+	os.RemoveAll(rootTmpDir)
+	cwd, _ := os.Getwd()
+
+	// Download swagger yaml and convert to JSON
+	d, err := fmts.YAMLDoc(swaggerURL)
+	if err != nil {
+		log.Fatalf("Failed to convert swagger yaml to json: %v", err)
+	}
+
+	var sw spec.Swagger
+	if err := json.Unmarshal(d, &sw); err != nil {
+		log.Fatalf("Failed to convert swagger yaml to json: %v", err)
+	}
+
+	version := sw.Info.Version
+	fmt.Printf("VERSION: %s\n", version)
+
+	var only string
+	if len(os.Args) > 1 && os.Args[1] != "" {
+		only = os.Args[1]
+	}
+
+	var languages []string
+	if only != "" {
+		languages = append(languages, only)
+	} else {
+		// Download available languages from swagger generator api
+		languages = getLanguages()
+	}
+
+	for _, language := range languages {
+		var skipFiles []string
+		tmpDir := filepath.Join(rootTmpDir, language)
+		srcDir := filepath.Join(tmpDir, "src")
+		clientDir := filepath.Join(tmpDir, fmt.Sprintf("%s-client", language))
+
+		var options map[string]interface{}
+		var deploy [][]string
+
+		// Specfic language configurations
+		switch language {
```

Comment:
Loop with a switch inside. It seems we'd be better use a map here, instead of testing for all languages against the subset that the switch block represents.